packages/flutter/lib/src/cupertino/text_selection.dart

```diff
@@ -62,6 +63,155 @@ const TextStyle _kToolbarButtonDisabledFontStyle = TextStyle(
 // Eyeballed value.
 const EdgeInsets _kToolbarButtonPadding = EdgeInsets.symmetric(vertical: 10.0, horizontal: 18.0);
 
+// Generates the child that's passed into CupertinoTextSelectionToolbar.
+class _CupertinoTextSelectionToolbarWrapper extends StatefulWidget {
+  const _CupertinoTextSelectionToolbarWrapper({
+    Key key,
+    this.barTopY,
+    this.arrowTipX,
+    this.handleCut,
+    this.handleCopy,
+    this.handlePaste,
+    this.handleSelectAll,
+    this.isArrowPointingDown,
+  }) : super(key: key);
+
+  final double arrowTipX;
+  final double barTopY;
+  final VoidCallback handleCut;
+  final VoidCallback handleCopy;
+  final VoidCallback handlePaste;
+  final VoidCallback handleSelectAll;
+  final bool isArrowPointingDown;
+
+  @override
+  _CupertinoTextSelectionToolbarWrapperState createState() => _CupertinoTextSelectionToolbarWrapperState();
+}
+
+class _CupertinoTextSelectionToolbarWrapperState extends State<_CupertinoTextSelectionToolbarWrapper> with WidgetsBindingObserver {
+  // Whether the user's system clipboard is empty as of the last check.
+  bool _isClipboardEmpty;
+
+  // Asynchronously check if the clipboard has data and update
+  // _isClipboardEmpty if needed.
+  void _updateClipboardStatus() {
+    Clipboard.getData(Clipboard.kTextPlain).then((ClipboardData value) {
+      if (!mounted) {
+        return;
+      }
+      final bool isClipboardEmpty = value?.text == null || value.text.isEmpty;
+      if (isClipboardEmpty == _isClipboardEmpty) {
+        return;
+      }
+      setState(() {
+        _isClipboardEmpty = isClipboardEmpty;
+      });
+    });
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    WidgetsBinding.instance.addObserver(this);
+    if (widget.handlePaste != null) {
+      _updateClipboardStatus();
+    }
+  }
+
+  @override
+  void didUpdateWidget(_CupertinoTextSelectionToolbarWrapper oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (widget.handlePaste != null) {
+      _updateClipboardStatus();
+    }
+  }
+
+  @override
+  void didChangeAppLifecycleState(AppLifecycleState state) {
+    switch (state) {
+      case AppLifecycleState.resumed:
+        if (widget.handlePaste != null) {
+          _updateClipboardStatus();
+        }
+        break;
+      case AppLifecycleState.detached:
+        break;
+      case AppLifecycleState.inactive:
+        break;
+      case AppLifecycleState.paused:
+        break;
```

Comment:
nit: this can be more compact. 

```suggestion
      case AppLifecycleState.detached:
      case AppLifecycleState.detached:
      case AppLifecycleState.paused:
         // Nothing to do.
```